### PR TITLE
Adding @reta to OpenSearch maintainers.

### PR DESCRIPTION
**Pull request by dblock**
Signed-off-by: dblock <dblock@amazon.com>

### Description

Following the process in https://github.com/opensearch-project/.github/pull/59, I had nominated Andriy Redko (@reta) as a co-maintainer. The maintainers have voted and agreed to this nomination.
 
Andriy is the author of [124 PRs](https://github.com/opensearch-project/OpenSearch/pulls?q=is%3Apr+author%3Areta+is%3Aclosed) into OpenSearch, [second](https://github.com/opensearch-project/OpenSearch/graphs/contributors?from=2021-05-27&to=2022-04-04&type=c) in code change volume since the fork.
 
image
 
Some of reta@’s notable early code contributions were in the supporting infrastructure space, and have increasingly been shifting into adding new features:
 
- https://github.com/opensearch-project/OpenSearch/pull/1500: Concurrent searching
- https://github.com/opensearch-project/OpenSearch/pull/1521, [1302](https://github.com/opensearch-project/OpenSearch/pull/1302), [1390](https://github.com/opensearch-project/OpenSearch/pull/1390), [1789](https://github.com/opensearch-project/OpenSearch/pull/1789): Improving/upgrading/fixing repository-azure
- https://github.com/opensearch-project/OpenSearch/pull/1771, [1721](https://github.com/opensearch-project/OpenSearch/pull/1721), …: Upgrading log4j
- https://github.com/opensearch-project/OpenSearch/pull/1609, [1803](https://github.com/opensearch-project/OpenSearch/pull/1803), [2078](https://github.com/opensearch-project/OpenSearch/pull/2078), …: Upgrades to Gradle
- https://github.com/opensearch-project/OpenSearch/pull/1358. [1803](https://github.com/opensearch-project/OpenSearch/pull/1803), [1476](https://github.com/opensearch-project/OpenSearch/pull/1476), [2007](https://github.com/opensearch-project/OpenSearch/pull/2007): Modernizing, consolidating, upgrading JDK
- https://github.com/opensearch-project/OpenSearch/pull/2001: Support for OPENSEARCH_JAVA_HOME
 
Andriy is active almost daily in the project, thoughtfully reviewing and commenting on pull requests ([example](https://github.com/opensearch-project/OpenSearch/pull/2639#discussion_r839909012)). He also has write permissions on https://github.com/opensearch-project/opensearch-plugin-template-java/ already. 

This PR also sorts current maintainers alphabetically.  
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by VachaShah — APPROVED** [review]
This is awesome!

**Review by owaiskazi19 — APPROVED** [review]
Welcome to the club, @reta!

**Review by kartg — APPROVED** [review]
💯

**Review by andrross — APPROVED** [review]

File: MAINTAINERS.md
Comment (andrross) [inline MAINTAINERS.md:7]: While you're here can you add the missing 'r' in the link name? :)

**Review by saratvemulapalli — APPROVED** [review]

File: MAINTAINERS.md
Comment (nknize) [reply MAINTAINERS.md]: Done :)

**Review by nknize — APPROVED** [review]
LGTM

**Comment by dblock** [comment]
I've asked an org admin (@hyandell @CEHENKLE) to add @reta. Will merge when that's done.

**Review by mch2 — APPROVED** [review]

**Comment by reta** [comment]
So proud to be nominated, thank you so much team!! :heart: :heart: :heart: :heart:  I will continue to do my best but it is nearly impossible to chase @nknize :smile:  :smile:

**Comment by mattweber** [comment]
Congrats @reta!

**Comment by dblock** [comment]
Screen Shot 2022-04-15 at 11 10 00 AM

**Comment by nknize** [comment]
So happy to have you, @reta, as a maintainer! Some excellent contributions over the last months. Thanks for sticking around while the maintainer process was being figured out.

**Pull request merged**
